geo/spatial-geo-geonamesindex/src/main/java/org/codice/ddf/spatial/geo/indexer/GeoNamesIndexer.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+
+package org.codice.ddf.spatial.geo.indexer;
+
+import static org.apache.lucene.index.IndexWriter.MaxFieldLength;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.io.PrintStream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.Version;
+import org.codice.ddf.spatial.geo.GeoIndexer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+
+public class GeoNamesIndexer implements GeoIndexer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeoNamesIndexer.class);
+
+    private static final PrintStream CONSOLE = System.out;
+
+    private String indexLocation;
+
+    public void setIndexLocation(final String indexLocation) {
+        this.indexLocation = indexLocation;
+    }
+
+    @Override
+    public void updateIndex(final String fileLocation, final boolean create) {
+        final String inputFileLocation = getInputFileLocation(fileLocation);
+
+        if (inputFileLocation != null) {
+            buildIndex(inputFileLocation, create);
+        }
+    }
+
+    private String getInputFileLocation(final String fileLocation) {
```

Comment:
Now I'm thinking - why not a URL to a file on a webserver? :) - maybe a future ticket.